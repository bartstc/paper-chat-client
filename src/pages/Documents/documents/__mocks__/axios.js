module.exports = {
  get: () => {
    return Promise.resolve({
      data: [
        {
          id: 1,
          category: 'WORK',
          title: 'test'
        },
        {
          id: 2,
          category: 'HOBBY',
          title: 'test2'
        }
      ]
    });
  }
};
