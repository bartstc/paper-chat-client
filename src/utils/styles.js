import { createGlobalStyle } from 'styled-components';

export const color = {
  white: `#FFF`,
  black: `#151615`,
  grey: '#565656',
  accent: `#0075C4`,
};

export const fontFamily = {
  primary: `'Montserrat', sans-serif`,
};

export const fontWeight = {
  light: `300`,
  bold: `700`,
};

const size = {
  mobileS: '320px',
  mobileM: '375px',
  mobileL: '425px',
  tablet: '768px',
  laptop: '1024px',
  laptopL: '1440px',
  laptopXL: '1644px',
  desktop: '2560px',
};

export const device = {
  mobileS: `(min-width: ${size.mobileS})`,
  mobileM: `(min-width: ${size.mobileM})`,
  mobileL: `(min-width: ${size.mobileL})`,
  tablet: `(min-width: ${size.tablet})`,
  laptop: `(min-width: ${size.laptop})`,
  laptopL: `(min-width: ${size.laptopL})`,
  laptopXL: `(min-width: ${size.laptopXL})`,
  desktop: `(min-width: ${size.desktop})`,
  desktopL: `(min-width: ${size.desktop})`,
};

export const GlobalStyle = createGlobalStyle`
  body {
    color: ${color.black};
    font-family: ${fontFamily.primary};
    font-weight: ${fontWeight.light};
    overflow-x: hidden;
  }
`;

export const accent = `
  color: ${color.accent};
`;
