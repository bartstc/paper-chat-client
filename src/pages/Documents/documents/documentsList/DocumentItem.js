import React from 'react';
import axios from 'axios';
import { useDocumentsDispatch } from '../../../../context/documentsContext';
import { DocItem, DocTitle, Category, DeleteBtn } from './DocumentItem.styles';

const categoryColors = {
  WORK: '#a30402',
  HOBBY: '#47990c',
  STUDY: '#0779b3',
  TRAVEL: '#edd100',
  FINANCE: '#990c90'
};

const DocumentItem = ({ id, category, title, fetchDocuments }) => {
  const dispatch = useDocumentsDispatch();

  const deleteDocument = async id => {
    try {
      dispatch({ type: 'LOADING_START' });
      await axios.delete(`/documents/${id}`);
      fetchDocuments();
    } catch (err) {
      console.log(err.response.data);
    }
  };

  return (
    <DocItem key={id}>
      <Category color={categoryColors[category]} />
      <DocTitle to={`/editor/${id}`}>{title}</DocTitle>
      <DeleteBtn onClick={() => deleteDocument(id)}>X</DeleteBtn>
    </DocItem>
  );
};

export default DocumentItem;
