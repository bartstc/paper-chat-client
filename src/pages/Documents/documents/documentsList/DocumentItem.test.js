import React from 'react';
import { mount } from 'enzyme';
import axios from 'axios';
import toJson from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';
import {
  DocumentsStateContext,
  DocumentsDispatchContext
} from '../../../../context/documentsContext';
import DocumentItem from './DocumentItem';
import { DeleteBtn } from './DocumentItem.styles';

let state = {
  documents: [
    {
      id: 1,
      category: 'WORK',
      title: 'test'
    },
    {
      id: 2,
      category: 'HOBBY',
      title: 'test2'
    }
  ],
  loading: false
};

let dispatch = jest.fn();

let DocumentsProvider = ({ children }) => {
  return (
    <DocumentsStateContext.Provider value={state}>
      <DocumentsDispatchContext.Provider value={dispatch}>
        {children}
      </DocumentsDispatchContext.Provider>
    </DocumentsStateContext.Provider>
  );
};

describe('<DocumentItem />', () => {
  const mockFetchDocuments = jest.fn();
  let props = {
    id: 1,
    title: 'test',
    category: 'WORK',
    fetchDocuments: mockFetchDocuments
  };

  let wrapper = mount(
    <Router>
      <DocumentsProvider>
        <DocumentItem {...props} />
      </DocumentsProvider>
    </Router>
  );

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('contains DeleteBtn', () => {
    expect(wrapper.find(DeleteBtn).length).toEqual(1);
  });

  // describe('when the `DeleteBtn` is clicked', () => {
  // beforeEach(() => {
  //   wrapper = mount(
  //     <Router>
  //       <DocumentsProvider>
  //         <DocumentItem {...props} />
  //       </DocumentsProvider>
  //     </Router>
  //   );
  //   wrapper.find(DeleteBtn).simulate('click');
  // });

  // it('should made a delete request', () => {
  //   const deleteSpy = jest.spyOn(axios, 'delete');
  //   expect(deleteSpy).toBeCalled();
  // });

  // it('call dispatch function with type `LOADING_START`', () => {
  //   const dispatch = jest.fn();
  //   expect(dispatch).toBeCalledWith({ type: 'LOADING_START' });
  // });

  // it('refetch documents', () => {
  //   expect(mockFetchDocuments).toHaveBeenCalled();
  // });
  // });
});
