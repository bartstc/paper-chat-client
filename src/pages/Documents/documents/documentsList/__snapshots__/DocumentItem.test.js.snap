// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentItem /> renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DocumentsProvider>
      <DocumentItem
        category="WORK"
        fetchDocuments={[MockFunction]}
        id={1}
        title="test"
      >
        <styled.li
          key="1"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "eGOcHt",
                  "rules": Array [
                    "
  position: relative;
  margin-bottom: 0.8em;
  height: 40px;
  display: flex;
  align-items: center;
",
                  ],
                },
                "displayName": "styled.li",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "li",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <li
              className="sc-bdVaJa eGOcHt"
            >
              <styled.span
                color="#a30402"
              >
                <StyledComponent
                  color="#a30402"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bwzfXH",
                        "isStatic": false,
                        "lastClassName": "jWulDI",
                        "rules": Array [
                          "
  display: block;
  height: 21px;
  width: 45px;
  background: ",
                          [Function],
                          ";
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bwzfXH",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="sc-bwzfXH jWulDI"
                    color="#a30402"
                  />
                </StyledComponent>
              </styled.span>
              <Styled(Link)
                to="/editor/1"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": true,
                        "lastClassName": "cGUhbJ",
                        "rules": Array [
                          "
  padding: 0 1.4em 0 0.8em;
",
                        ],
                      },
                      "displayName": "Styled(Link)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  to="/editor/1"
                >
                  <Link
                    className="sc-htpNat cGUhbJ"
                    to="/editor/1"
                  >
                    <a
                      className="sc-htpNat cGUhbJ"
                      href="/editor/1"
                      onClick={[Function]}
                    >
                      test
                    </a>
                  </Link>
                </StyledComponent>
              </Styled(Link)>
              <styled.button
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": true,
                        "lastClassName": "cDTDlc",
                        "rules": Array [
                          "
  width: 20px;
  height: 20px;
  position: absolute;
  top: 10px;
  right: 0;
  border: none;
  color: ",
                          "#FFF",
                          ";
  font-weight: ",
                          "700",
                          ";
  font-size: 0.85rem;
  background: #a10000;
  cursor: pointer;
",
                        ],
                      },
                      "displayName": "styled.button",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "button",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <button
                    className="sc-bxivhb cDTDlc"
                    onClick={[Function]}
                  >
                    X
                  </button>
                </StyledComponent>
              </styled.button>
            </li>
          </StyledComponent>
        </styled.li>
      </DocumentItem>
    </DocumentsProvider>
  </Router>
</BrowserRouter>
`;
